alicloud/resource_alicloud_db_database.go

```diff
@@ -96,77 +94,73 @@ func resourceAlicloudDBDatabaseCreate(d *schema.ResourceData, meta interface{})
 	if d.HasChange("description") && !d.IsNewResource() {
-		parts := strings.Split(d.Id(), COLON_SEPARATED)
+		parts, err := ParseResourceId(d.Id(), 2)
+		if err != nil {
+			return WrapError(err)
+		}
 		request := rds.CreateModifyDBDescriptionRequest()
 		request.DBInstanceId = parts[0]
 		request.DBName = parts[1]
 		request.DBDescription = d.Get("description").(string)
-
-		_, err := client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
+		var raw interface{}
+		raw, err = client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
 			return rdsClient.ModifyDBDescription(request)
 		})
 		if err != nil {
-			return fmt.Errorf("ModifyDatabaseDescription got an error: %#v", err)
+			return WrapError(err)
 		}
-		d.SetPartial("description")
+		addDebug(request.GetActionName(), raw)
 	}
-
-	d.Partial(false)
 	return resourceAlicloudDBDatabaseRead(d, meta)
 }
 
 func resourceAlicloudDBDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*connectivity.AliyunClient)
-	rsdService := RdsService{client}
-	parts := strings.Split(d.Id(), COLON_SEPARATED)
+	rdsService := RdsService{client}
+	parts, err := ParseResourceId(d.Id(), 2)
+	if err != nil {
+		return WrapError(err)
+	}
 	request := rds.CreateDeleteDatabaseRequest()
 	request.DBInstanceId = parts[0]
 	request.DBName = parts[1]
 	// wait instance status is running before deleting database
-	if err := rsdService.WaitForDBInstance(parts[0], Running, 1800); err != nil {
+	if err := rdsService.WaitForDBInstance(parts[0], Running, 1800); err != nil {
 		return WrapError(err)
 	}
-	return resource.Retry(5*time.Minute, func() *resource.RetryError {
-		_, err := client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
-			return rdsClient.DeleteDatabase(request)
-		})
-		if err != nil {
-			if rsdService.NotFoundDBInstance(err) || IsExceptedError(err, InvalidDBNameNotFound) {
-				return nil
-			}
-			return resource.RetryableError(fmt.Errorf("Delete database %s timeout and got an error: %#v.", parts[1], err))
-		}
-
-		if _, err := rsdService.DescribeDatabaseByName(parts[0], parts[1]); err != nil {
-			if NotFoundError(err) {
-				return nil
-			}
-			return resource.NonRetryableError(fmt.Errorf("Error Describe DB InstanceAttribute: %#v", err))
+	raw, err := client.WithRdsClient(func(rdsClient *rds.Client) (interface{}, error) {
+		return rdsClient.DeleteDatabase(request)
+	})
+	if err != nil {
+		if rdsService.NotFoundDBInstance(err) || IsExceptedError(err, InvalidDBNameNotFound) {
+			return nil
 		}
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
 
-		return resource.RetryableError(fmt.Errorf("Delete database %s timeout.", parts[1]))
-	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	return rdsService.WaitForDBDatabase(d.Id(), Deleted, DefaultTimeoutMedium)
```

Comment:
line 165 should be` return WrapError(rdsService.WaitForDBDatabase(d.Id(), Deleted, DefaultTimeoutMedium))`